content/sensu-go/5.17/api/secrets.md

```diff
@@ -0,0 +1,386 @@
+
+HTTP/1.1 204 No Content
+{{< /highlight >}}
+
+#### API Specification {#providers-provider-delete-specification}
+
+/providers/:provider (DELETE) | 
+--------------------------|------
+description               | Deletes the specified provider from Sensu.
+example url               | http://hostname:8080/api/enterprise/secrets/v1/providers/env
+response codes            | <ul><li>**Success**: 204 (No Content)</li><li>**Missing**: 404 (Not Found)</li><li>**Error**: 500 (Internal Server Error)</li></ul>
+
+## The `/secrets` endpoint
+
+### `/secrets` (GET)
+
+The `/secrets` API endpoint provides HTTP GET access to a list of secrets.
+
+#### EXAMPLE {#secrets-get-example}
+
+The following example demonstrates a request to the `/secrets` API endpoint, resulting in a list of secrets for the specified namespace.
+
+{{< highlight shell >}}
+curl -X GET \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/namespaces/default/secrets \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN"
+
+HTTP/1.1 200 OK
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
+{{< /highlight >}}
+
+#### API Specification {#secrets-get-specification}
+
+/secrets (GET)  | 
+---------------|------
+description    | Returns the list of secrets for the specified namespace. `id` is the identifying key for the provider to retrieve the secret. `provider` is the name of the provider with the secret.
+example url    | http://hostname:8080/api/enterprise/secrets/v1/namespaces/default/secrets
+response type  | Array
+response codes | <ul><li>**Success**: 200 (OK)</li><li>**Error**: 500 (Internal Server Error)</li></ul>
+output         | {{< highlight shell >}}
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
+{{< /highlight >}}
+
+## The `/secrets/:secret` endpoint {#the-secrets-secret-endpoint}
+
+### `/secrets/:secret` (GET) {#secrets-secret-get}
+
+The `/secrets/:secret` API endpoint provides HTTP GET access to data for a specific `secret`, by secret name.
+
+#### EXAMPLE {#secrets-secret-get-example}
+
+In the following example, querying the `/secrets/:secret` API endpoint returns a JSON map that contains the requested `:secret`.
+
+{{< highlight shell >}}
+curl -X GET \
+http://127.0.0.1:8080/api/enterprise/secrets/v1/namespaces/default/secrets/sensu-ansible-token \
+-H "Authorization: Bearer $SENSU_ACCESS_TOKEN"
+
+HTTP/1.1 200 OK
+[
+  {
+    "type": "Secret",
+    "api_version": "secrets/v1",
+    "metadata": {
+      "name": "sensu-ansible-token",
+      "namespace": "default"
+    },
+    "spec": {
+      "id": "ANSIBLE_TOKEN",
+      "provider": "env"
+    }
+  }
+]
```

Comment:
When getting a single secret `/secrets/:secret` it should be a single JSON object, not a JSON list of objects.
```
{
    "type": "Secret",
    "api_version": "secrets/v1",
    "metadata": {
        "name": "sensu-ansible-token",
        "namespace": "default"
    },
    "spec": {
        "id": "ANSIBLE_TOKEN",
        "provider": "env"
    }
}
```